### DISCUSSION TITLE
Allow instances for asset fields

### DISCUSSION THREAD

**Discussion by foxtrotcz** [craftcms/cms#14728] [Ideas] [answered]
If possible asset fields should act the same as other fields and be reusable with instances.
Thanks.

**Comment by mini71** [comment]
Multi instance fields are extremly useful. Dear @brandonkelly , any chance that this will be possible for assets fields?

**Reply by brandonkelly** [reply]
It’s theoretically possible, but will take some effort, so didn’t make it into 5.0.

**Reply by mikesnoeren** [reply]
Not sure about the technical implications this would bring, but this feature would be very much appreciated! Currently it feels like the modular way of setting up the CMS only works for 70%, because of all the double relational fields. 

In addition to that, would it be possible to have instanced fields bring their own settings? Kind of like how classes can extend eachother? That way we could alter field settings on instance level while creating entry types..

**Reply by brandonkelly** [reply]
@mikesnoeren That has been suggested before. See https://github.com/craftcms/cms/discussions/8688#discussioncomment-2811602

**Reply by droemmelbert** [reply]
@brandonkelly I just want to emphasize that this feature request by @FoxtrotCZ is something I would also be very happy to see in the future!

**Comment by brandonkelly** [comment, marked as answer]
Coming in Craft 5.3 (#15400) 🎉

**Reply by brandonkelly** [reply]
Craft 5.3.0-beta.1 is out with support for multi-instance relational fields 🎉 (#15400)

We’re targeting August 6 for the GA release. If you’d like to test ahead of time, change your `craftcms/cms` requirement in composer.json to:

```json
"craftcms/cms": "^5.3.0-beta.1",
```

and then run `composer update`.
